apps/studio/src/server/modules/site/__tests__/site.router.test.ts

```diff
@@ -340,6 +378,394 @@ describe("site.router", async () => {
     })
   })
 
+  describe("updateSiteConfig", () => {
+    it("should throw 401 if not logged in", async () => {
+      // Arrange
+      const unauthedSession = applySession()
+      const unauthedCaller = createCaller(createMockRequest(unauthedSession))
+
+      // Act
+      const result = unauthedCaller.updateSiteConfig({
+        siteName: MOCK_SITE_NAME,
+        logoUrl: MOCK_LOGO_URL,
+        url: "www.isomer.gov.sg",
+        theme: "isomer-next",
+        siteId: 1,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "UNAUTHORIZED" }),
+      )
+    })
+    it("should throw 403 if the user does not have write access to the site", async () => {
+      // Arrange
+      const { site } = await setupSite()
+
+      // Act
+      const result = caller.updateSiteConfig({
+        siteName: MOCK_SITE_NAME,
+        logoUrl: MOCK_LOGO_URL,
+        url: "www.isomer.gov.sg",
+        theme: "isomer-next",
+        siteId: site.id,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "FORBIDDEN",
+          message:
+            "You do not have sufficient permissions to perform this action",
+        }),
+      )
+    })
+    it("should update the site config if the user is a site admin", async () => {
+      // Arrange
+      const { site } = await setupSite()
+      await setupAdminPermissions({
+        userId: session.userId,
+        siteId: site.id,
+      })
+
+      // Act
+      const result = await caller.updateSiteConfig({
+        siteName: MOCK_SITE_NAME,
+        logoUrl: MOCK_LOGO_URL,
+        url: "www.isomer.gov.sg",
+        theme: "isomer-next",
+        siteId: site.id,
+      })
+
+      // Assert
+      expect(result).toEqual({
+        siteName: MOCK_SITE_NAME,
+        logoUrl: MOCK_LOGO_URL,
+        url: "www.isomer.gov.sg",
+        theme: "isomer-next",
+      })
+    })
+    it("should generate an audit log entry", async () => {
```

Comment:
nit: we should assert this together with the previous test, meaning we combine the two tests

Reply:
i actually did this cos it's clearer what's being asserted. for discussion going forward, should we do this or combine all the assertions for success case? 

i thought having various assertions for the invariants in teh success case made it clearer. i'm happy to change it to whichever the team prefers
@adriangohjw @dcshzj

Reply:
discussed on slack: https://opengovproducts.slack.com/archives/C06R4DX966P/p1762933350084159?thread_ts=1762916421.142049&cid=C06R4DX966P; 

not doing this (sorry)